### PR TITLE
slack bot

### PR DESCRIPTION
**Pull request by ronuchit**
Instructions for @tomsilver: https://docs.google.com/document/d/1eeLkxl6Nbb4dvG4gp7wdBS_GR6bOIrbH1WSRMz4ag3U/edit?usp=sharing

### CODE REVIEW

File: launch_slack_bot.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+               "- `cs <any string to search on our github repo>`, e.g., "
+               "`cs def flush_cache`\n"
+               "- `tom`\n")
+        return [ret]
+
+    def get_filename(self) -> Optional[str]:
+        return None
+
+
+class TomEmojiResponse(Response):
```
Comment (tomsilver) [inline launch_slack_bot.py:104]: 🥇

File: launch_slack_bot.py [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+        commands = ["rm -f output.txt"]
+        for command in self._get_commands():
+            # 2) SSH into the user's login server.
+            commands.append(
+                f"ssh {user}@{SUPERCLOUD_LOGIN_SERVER} "
+                # 3) Change into the desired directory and source
+                # the user's conda environment.
+                f"'cd {dir_name} && source {CONDA} {conda_env} "
+                # 4) This step is tricky! For some reason, when you
+                # SSH, the PYTHONPATH isn't preserved, so we update
```
Comment (tomsilver) [inline launch_slack_bot.py:197]: interesting... good job figuring that out

**Review by tomsilver — APPROVED** [review]
incredible! thank you!!!

one question: is it possible to summon the slackbot outside the special channel? I'm thinking about the use case you mentioned where you want to show someone code in conversation.

**Comment by ronuchit** [comment]
yup, will add instructions to the doc

**Pull request merged**